src/briefcase/integrations/download.py

```diff
@@ -97,3 +87,43 @@ def file(self, url, download_path, role=None):
             raise NetworkFailure(f"download {description}") from e
 
         return filename
+
+    def _fetch_and_write_content(self, response, filename):
+        """Write the content from the requests Response to file.
+
+        The data is initially written in to a temporary file in the Briefcase
+        cache. This avoids partially downloaded files masquerading as complete
+        downloads in later Briefcase runs. The temporary file is only moved
+        to ``filename`` if the download is successful; otherwise, it is deleted.
+
+        :param response: ``requests.Response``
+        :param filename: full filesystem path to save data
+        """
+        temp_file = tempfile.NamedTemporaryFile(
+            dir=filename.parent,
+            prefix=f"{filename.name}.",
+            suffix=".download",
+            delete=False,
+        )
```

Comment:
By default, `tempfile` uses random chars for the filename; prefixing with the intended filename makes it possible to know what the temporary data is and the suffix makes it easy to identity it as a temporary file.

Here's an example filename:
```
Python-3.10-linux-x86_64-support.b4.tar.gz.vdqfwdt2.download
```